tests/pretty/autodiff_forward.rs

```diff
@@ -36,4 +36,22 @@ pub fn f5(x: &[f64], y: f64) -> f64 {
     unimplemented!()
 }
 
+struct DoesNotImplDefault;
+#[autodiff(df6, Forward, Const)]
+pub fn f6() -> DoesNotImplDefault {
+    unimplemented!()
+}
+
+// Make sure, that we add the None for the default return.
+#[autodiff(df7, Forward, Const)]
+pub fn f7(x: f32) -> () {}
+
+#[autodiff(f8_1, Forward, Dual, DualOnly)]
+#[autodiff(f8_2, Forward, 4, Dual, DualOnly)]
+#[autodiff(f8_3, Forward, 4, Dual, Dual)]
```

Comment:
I think we have reached the point where it may make more sense to have named fields instead of just a list of args

Reply:
not in this PR tho ^^

Reply:
Not fully opposed, but that's the point where I usually handwave and mumble something about user wrappers ^^
I'm not really convinced that my designs have the best UX, so I mostly care about exposing all features, making it safe and fast, and then let normal proc macros wrap this. Once users agreed on their favorite wrapper, we could then talk about upstreaming it into rustc.